packages/database/e2e/database.modular.e2e.js

```diff
@@ -0,0 +1,464 @@
+        } catch (error) {
+          error.message.should.containEql("'bytes' must be a number value");
+          return Promise.resolve();
+        }
+      });
+
+      it('throws if bytes is less than 1MB', async function () {
+        try {
+          firebase.database().setPersistenceCacheSizeBytes(1234);
+          return Promise.reject(new Error('Did not throw an Error.'));
+        } catch (error) {
+          error.message.should.containEql("'bytes' must be greater than 1048576 bytes (1MB)");
+          return Promise.resolve();
+        }
+      });
+
+      it('throws if bytes is greater than 10MB', async function () {
+        try {
+          firebase.database().setPersistenceCacheSizeBytes(100000000000000);
+          return Promise.reject(new Error('Did not throw an Error.'));
+        } catch (error) {
+          error.message.should.containEql("'bytes' must be less than 104857600 bytes (100MB)");
+          return Promise.resolve();
+        }
+      });
+
+      it('calls setPersistenceCacheSizeBytes successfully', async function () {
+        firebase.database().setPersistenceCacheSizeBytes(1048576); // 1mb
+      });
+    });
+
+    describe('getServerTime()', function () {
+      it('returns a valid date', async function () {
+        const date = firebase.database().getServerTime();
+        date.getDate.should.be.Function();
+      });
+    });
+
+    describe('handles invalid references()', function () {
+      it('returns a valid date', async function () {
+        try {
+          firebase.database().ref('$');
+          return Promise.reject(new Error('Did not throw an Error.'));
+        } catch (error) {
+          error.message.should.containEql(
+            "Paths must be non-empty strings and can't contain #, $, [, ], ' or ?",
+          );
+          return Promise.resolve();
+        }
+      });
+    });
+  });
+
+  describe('modular', function () {
+    describe('namespace', function () {
+      it('accessible from getDatabase', function () {
+        const { getDatabase } = databaseModular;
+
+        const app = firebase.app();
+        const database = getDatabase();
+        should.exist(app.database);
+        database.app.should.eql(app);
+      });
+
+      it('supports multiple apps', async function () {
+        const { getDatabase } = databaseModular;
+        const database = getDatabase();
+        const secondaryDatabase = getDatabase(firebase.app('secondaryFromNative'));
+
+        database.app.name.should.eql('[DEFAULT]');
+
+        secondaryDatabase.app.name.should.eql('secondaryFromNative');
+
+        firebase.app('secondaryFromNative').database().app.name.should.eql('secondaryFromNative');
+      });
+    });
+
+    it.skip('supports custom database URL', async function () {
+      // FIXME: the v8 API doesn't seem to have a URL argument
+      throw new Error('Incomplete test');
+
+      // Below is the v8 test, which has nothing to do with custom database URLs
```

Comment:
Can't you pass in a custom URL like this:

```js
const reference = firebase
  .app()
  .database('https://<databaseName>.<region>.firebasedatabase.app/')
  .ref('/users/123');
```

Reply:
Ah I missed this test. But question. The original v8 test is the following:

https://github.com/invertase/react-native-firebase/blob/0716a2a73bed50155783c772bcae5c4971966518/packages/database/e2e/database.modular.e2e.js#L38-L46

As per my comment after the FIXME, this test has nothing to do with URLs. How should I test this, and should I update the v8 test too?

Reply:
Yeah, the test is incorrectly named. But to get a db version for a specific app, you use the first arg of `getDatabase()`.

```js
getDatabase(app?: FirebaseApp, url?: string):
```

Update the name of the test as well. Not sure if this is a test you can port over to be honest, I think there is only one way to get a different db instance with modular. I think you have to pass it in to `getDatabase(app?: FirebaseApp, url?: string):` so I think you can just remove this test for modular. Double check first as my cursory look over the docs suggest there is only one way to access.

Reply:
So there is a test right above this test that is the exact same code. I think this was an oversight while writing tests. The code was copied, the test renamed, but the code never changed. See:

https://github.com/invertase/react-native-firebase/blob/0716a2a73bed50155783c772bcae5c4971966518/packages/database/e2e/database.modular.e2e.js#L27-L46

Should I remove this custom URL test altogether, from v8 and modular both? I'm not sure how to test the URL connection honestly. Don't think the emulator supports multiple connections, so we'd have to test it on an actual FB project?

Reply:
Yes, correct. That test can just be removed as it is the exact same as the one above 👍